skimage/io/sift.py

```diff
@@ -51,10 +59,13 @@ def _sift_read(f, mode='SIFT'):
         datatype = np.dtype([('column', float), ('row', float),
                              ('second_moment', (float, 3)),
                              ('sign', float), ('data', (float, feature_len))])
+
     data = np.fromfile(f, sep=' ')
     if data.size != nr_features * datatype.itemsize / np.dtype(float).itemsize:
         raise IOError("Invalid %s feature file." % mode)
 
+    f.close()
```

Comment:
@sciunto see https://github.com/stefanv/scikit-image/blob/11a26ffe57363daf62364df5635e38af82cd6867/skimage/io/sift.py#L47-L48. I'd say that it should there, and only if `f` is not a valid filename.